internal/uvm/create_wcow.go

```diff
@@ -69,6 +69,15 @@ func CreateWCOW(ctx context.Context, opts *OptionsWCOW) (_ *UtilityVM, err error
 		vsmbDirShares:       make(map[string]*VSMBShare),
 		vsmbFileShares:      make(map[string]*VSMBShare),
 	}
+
+	if !opts.AllowOvercommit && opts.EnableDeferredCommit {
```

Comment:
I think we should fail earlier than when we're actually creating the pod. I envisioned a func called at the end of say SpecToUVMCreateOpts that validates the resulting options set. This also makes it easier for us to add onto validate opts constraints in the future rather than muddying up the create_*cow funcs.